apps/src/code-studio/components/Libraries/LibraryCreationDialog.jsx

```diff
@@ -119,16 +126,17 @@ class LibraryCreationDialog extends React.Component {
             style={styles.textarea}
             placeholder="Write a description of your library"
           />
-          {this.state.selectedFunctionList.map(selectedFunction => {
-            let name = selectedFunction.functionName;
-            let comment = selectedFunction.comment;
+          {this.state.sourceFunctionList.map(sourceFunction => {
+            let name = sourceFunction.functionName;
+            let comment = sourceFunction.comment;
             return (
               <div key={keyIndex++} style={styles.functionItem}>
                 <input
                   type="checkbox"
                   style={styles.largerCheckbox}
                   disabled={comment.length === 0}
                   onClick={this.validateInput}
+                  value={keyIndex}
```

Comment:
What does this do? (and why is the default -1?)

Reply:
Good question! This was a workaround for handling functions that have the same name/comments. Since we can't rely on unique values to connected the selected checkbox to the list of functions, I've shifted to using indices. Because javascript .map goes over the array items in order, we can assign each item created an index so that we can then track that index back to the function it is meant to represent.

Reply:
It is initialized at `-1` because it increments on each item created, so it has to start at `-1` so that it starts zero indexed. I don't love this solution, but I couldn't really figure out another way to do while still using a map. I'm open to suggestions!

Reply:
Don't love that it starts at -1?
You should be able to increment the counter at the end of .map instead of the beginning of .map. Then you can start it at 0.